redisvl/index/index.py

```diff
@@ -864,106 +894,101 @@ async def from_existing(
             ) from e
 
+        if redis_client is None:
+            raise ValueError(
+                "Failed to obtain a valid Redis client. "
+                "Please provide a valid redis_client or redis_url."
+            )
+
         # Fetch index info and convert to schema
         index_info = await cls._info(name, redis_client)
         schema_dict = convert_index_info_to_schema(index_info)
         schema = IndexSchema.from_dict(schema_dict)
-        index = cls(schema, **kwargs)
-        await index.set_client(redis_client)
-        return index
+        return cls(schema, redis_client=redis_client, **kwargs)
 
     @property
     def client(self) -> Optional[aredis.Redis]:
         """The underlying redis-py client object."""
         return self._redis_client
 
+    @deprecated_function("connect", "Pass connection parameters in __init__.")
     async def connect(self, redis_url: Optional[str] = None, **kwargs):
-        """Connect to a Redis instance using the provided `redis_url`, falling
-        back to the `REDIS_URL` environment variable (if available).
-
-        Note: Additional keyword arguments (`**kwargs`) can be used to provide
-        extra options specific to the Redis connection.
-
-        Args:
-            redis_url (Optional[str], optional): The URL of the Redis server to
-                connect to. If not provided, the method defaults to using the
-                `REDIS_URL` environment variable.
-
-        Raises:
-            redis.exceptions.ConnectionError: If the connection to the Redis
-                server fails.
-            ValueError: If the Redis URL is not provided nor accessible
-                through the `REDIS_URL` environment variable.
-
-        .. code-block:: python
-
-            index.connect(redis_url="redis://localhost:6379")
-
-        """
-        client = RedisConnectionFactory.connect(
-            redis_url=redis_url, use_async=True, **kwargs
+        """[DEPRECATED] Connect to a Redis instance. Use connection parameters in __init__."""
+        warnings.warn(
+            "connect() is deprecated; pass connection parameters in __init__",
+            DeprecationWarning,
         )
-        return await self.set_client(client)
-
-    @setup_async_redis()
-    async def set_client(self, redis_client: aredis.Redis):
-        """Manually set the Redis client to use with the search index.
-
-        This method configures the search index to use a specific
-        Async Redis client. It is useful for cases where an external,
-        custom-configured client is preferred instead of creating a new one.
-
-        Args:
-            redis_client (aredis.Redis): An Async Redis
-                client instance to be used for the connection.
-
-        Raises:
-            TypeError: If the provided client is not valid.
-
-        .. code-block:: python
+        # TODO: Intentionally not including required modules to match existing
```

Comment:
Ditto to above -- this did validate modules by running through the decorator used on `set_client`